ComplementaryScripts/modelCuration/mapKEGGID.m

```diff
@@ -0,0 +1,275 @@
+            model.metKEGGID(idx(i)) = xref_metMetaCyc(i,5);
+        end
+    end
+else
+    xref_metMetaCyc(:) = []; %erase all cell array, since no KEGGID matched and added
+end
+
+%map metMetaNetXID to metSEEDID via function mapIDsViaMNXref, then map metSEEDID to metKEGGID
+xref_metSEED = mapIDsViaMNXref('mets',model.metMetaNetXID,'MetaNetX','SEED');
+xref_metSEED(:,2) = model.mets; %match metSEEDID with model.mets
+xref_metSEED(:,3) = model.metMetaNetXID; %match metSEEDID with model.metMetaNetXID
+xref_metSEED(:,4) = model.metKEGGID; %match metSEEDID with model.metKEGGID
+empties = find(cellfun('isempty',xref_metSEED(:,1)));
+xref_metSEED(empties,:) = [];
+
+temp = what('ModelSEEDDatabase\Biochemistry'); %require cloned/downloaded GitHub repo ModelSEEDDatabase
+cd(temp.path);
+fid2 = fopen('compounds.tsv');
+format = repmat('%s ',1,21);
+format = strtrim(format);
+met_temp = textscan(fid2,format,'Delimiter','\t','HeaderLines',0);
+for i = 1:length(met_temp)
+    SEED_metInfo(:,i) = met_temp{i};
+end
+commentLines = startsWith(SEED_metInfo(:,1),'#');
+SEED_metInfo(commentLines,:) = [];
+fclose(fid2);
+
+%Input metKEGGID into xref array
+for i = 1:length(xref_metSEED)
+    if isempty(xref_metSEED{i,4})
+        idx = find(ismember(SEED_metInfo(:,1),xref_metSEED(i,1)));
+        if idx~=0
+            if ~isempty(SEED_metInfo{idx,19}) && contains(SEED_metInfo(idx,19),'KEGG')
+                temp_ID = extractBetween(SEED_metInfo(idx,19),'KEGG: ','|');
+                if isempty(temp_ID) %KEGGID may be present at the end
+                    temp_ID = extractAfter(SEED_metInfo(idx,19),'KEGG: ');
+                    temp_ID = erase(temp_ID,'"');
+                end
+                if contains(temp_ID,';') %if more than 1 ID recorded, take the 1st one by default
+                    temp_ID = split(temp_ID,';',2);
+                    temp_ID = temp_ID(1,1);
+                    warning('check %s, may contain more than 1 metKEGGID',string(SEED_metInfo(idx,1)));
+                end
+                xref_metSEED(i,5) = temp_ID;
+            end
+        end
+    end
+end
+
+if size(xref_metSEED,2) == 5
+    %only retain mapped metKEGGID if model.metKEGGID is empty
+    empties = find(cellfun('isempty',xref_metSEED(:,5)));
+    xref_metSEED(empties,:) = [];
+    empties = find(~cellfun('isempty',xref_metSEED(:,4)));
+    xref_metSEED(empties,:) = [];
+    
+    %add metKEGGID into model
+    [~,idx] = ismember(xref_metSEED(:,2),model.mets);
+    idx = idx(idx~=0);
+    
+    for i = 1:length(idx)
+        if ~contains(xref_metSEED(i,5),'G') %metKEGGID containing G not added, since it is not compatible with current SBML format
+            model.metKEGGID(idx(i)) = xref_metSEED(i,5);
+        end
+    end
+else
+    xref_metSEED(:) = []; %erase all cell array, since no KEGGID matched and added
+end
+
+%Save model
+temp = what('yeast-GEM\ComplementaryScripts');
```

Comment:
I suppose it is better to use relative path.  We should assume that people don't add yeastGEM to its Matlab Path. Besides, If using this one, it may cause conflict if I have multiple folders called `yeast-GEM\ComplementaryScripts`.  Please use relative path to cd back to the right folder.

Reply:
Resolved in b8a5d3283c45e15a654eb5e89e91570a7406ff61.